src/components/2faVerification.tsx

```diff
@@ -1,124 +1,134 @@
-"use client"
-import React, { useRef, useEffect, useState } from 'react'
```

Comment:
Always separate page logic from jsx elements using custom hooks or server actions.

Reply:
The logic in this file is not mine. I only modified it to use a global popup instead of creating a new one. All other logic in this file was originally created by Emmanuel.